test/e2e/multicast_test.go

```diff
@@ -0,0 +1,328 @@
+		if err != nil {
+			return false, err
+		}
+		if !mc.senderConfig.isHostNetwork {
+			if len(mrouteResult) == 0 {
+				return false, nil
+			}
+		} else {
+			if len(mrouteResult) != 0 {
+				return false, nil
+			}
+		}
+		// Check inbound multicast route and whether multicast interfaces has joined the multicast group.
+		for _, receiver := range mc.receiverConfigs {
+			for _, receiverMulticastInterface := range nodeMulticastInterfaces[receiver.nodeIdx] {
+				_, mRouteResult, _, err := RunCommandOnNode(nodeName(receiver.nodeIdx), fmt.Sprintf("ip mroute show to %s iif %s ", mc.group.String(), receiverMulticastInterface))
+				if err != nil {
+					return false, err
+				}
+				// If multicast traffic is sent from non-HostNetwork pods and senders-receivers are located in different nodes,
+				// the receivers should configure corresponding inbound multicast routes.
+				if mc.senderConfig.nodeIdx != receiver.nodeIdx && !receiver.isHostNetwork {
+					if len(mRouteResult) == 0 {
+						return false, nil
+					}
+				} else {
+					if len(mRouteResult) != 0 {
+						return false, nil
+					}
+				}
+				_, mAddrResult, _, err := RunCommandOnNode(nodeName(receiver.nodeIdx), fmt.Sprintf("ip maddr show %s | grep %s", receiverMulticastInterface, mc.group.String()))
+				if err != nil {
+					return false, err
+				}
+				// The receivers should also join multicast group.
+				// Note that in HostNetwork mode, the "join multicast" action is taken by mcjoin,
+				// which will not persist after mcjoin exits.
+				if !receiver.isHostNetwork {
+					if len(mAddrResult) == 0 {
+						return false, nil
+					}
+				} else {
+					if len(mAddrResult) != 0 {
+						return false, nil
+					}
+				}
+			}
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Error when waiting for multicast routes and stats: %v", err)
+	}
+	wg.Wait()
+}
+
+// computeMulticastInterfaces computes multicastInterfaces for each node.
+// It returns [][]string with its index as node index and value as multicastInterfaces for this node.
+func computeMulticastInterfaces(t *testing.T, data *TestData) ([][]string, error) {
+	nodes, err := testData.clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	workerIdx := 1
+	nodeIdxMcastIfaceMap := make(map[int][]string)
+	for _, node := range nodes.Items {
+		isControlPlaneNode := func() bool {
+			_, ok := node.Labels[labelNodeRoleControlPlane()]
+			return ok
+		}()
+		var nodeIdx int
+		if isControlPlaneNode {
+			nodeIdx = 0
+		} else {
+			nodeIdx = workerIdx
+			workerIdx++
+		}
+		_, localInterfacesStr, _, err := RunCommandOnNode(nodeName(nodeIdx), fmt.Sprintf("ls /sys/class/net"))
+		if err != nil {
+			return nil, err
+		}
+		multicastInterfaces, err := data.GetMulticastInterfaces(antreaNamespace)
```

Comment:
Move it to out of the loop as it's same for all Nodes.

Reply:
updated